### PR TITLE
[10] Add widget capability

### LINKED ISSUE #10: [Integration] As a user, I can add Widget to Home Screen
## Why

User wants to see Nimble e-commerce Widget on Screen.
 
## Acceptance Criteria

Screen Shot 2021-04-29 at 18 36 40

### PR DESCRIPTION
**Pull request by blyscuit**
resolves https://github.com/nimblehq/nimble-ecommerce-ios/issues/10

## What happened 👀

Add Widget Capability
 
## Insight 📝

Displaying widget from template

## Proof Of Work 📹

https://user-images.githubusercontent.com/6356137/117257354-fb922e00-ae75-11eb-956a-7014e4d22b9f.mp4

### CODE REVIEW

File: ecommerce-iosWidget/EcommerceiOSWidget.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        completion(entry)
+    }
+
+    func getTimeline(for configuration: ConfigurationIntent, in context: Context, completion: @escaping (Timeline<Entry>) -> Void) {
+        var entries: [SimpleEntry] = []
+
+        // Generate a timeline consisting of five entries an hour apart, starting from the current date.
+        let currentDate = Date()
+        for hourOffset in 0 ..< 5 {
+            let entryDate = Calendar.current.date(byAdding: .hour, value: hourOffset, to: currentDate)!
```
Comment (vnntsu) [inline ecommerce-iosWidget/EcommerceiOSWidget.swift]: Should we avoid force unwrapping?

File: ecommerce-iosWidget/EcommerceiOSWidget.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+}
+
+struct SimpleEntry: TimelineEntry {
+
+    let date: Date
+    let configuration: ConfigurationIntent
+}
+
+struct ecommerceiOSWidgetEntryView: View {
```
Comment (vnntsu) [inline ecommerce-iosWidget/EcommerceiOSWidget.swift]:
```suggestion
struct EcommerceiOSWidgetEntryView: View {
```

File: ecommerce-iosWidget/EcommerceiOSWidget.swift (hunk 1 above)
Comment (blyscuit) [reply ecommerce-iosWidget/EcommerceiOSWidget.swift]: Nice catch, we won't need this as it's pregen and can be removed.

File: ecommerce-iosWidget/EcommerceiOSWidget.swift (hunk 2 above)
Comment (vnntsu) [reply ecommerce-iosWidget/EcommerceiOSWidget.swift]: It couldn't be capital character or you forgot it?

**Review by vnntsu — APPROVED** [review]

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Pull request merged**